### PR TITLE
React to ForceSubtaskResults

### LINKED ISSUE #2188: [Concent] UC3 / Force Accept - as a Requestor, send `ForceSubtaskResultsResponse`
implement sending of `ForceSubtaskResultsResponse` to the Concent service in reaction to `ForceSubtaskResults`

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #2188

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -76,3 +76,4 @@ venv/*
 # dummy test temp files
 apps/dummy/test_tmp
 .python-version
+luxrender-output.png
```
Comment (shadeofblue) [inline .gitignore:79]: I was wondering whether we shouldn't rather fix the test not to use a file in the source tree ;p

File: golem/network/concent/received_handler.py [hunk 2]
```diff
@@ -99,6 +99,14 @@ def on_force_subtask_results_rejected(msg):
     logger.warning("[CONCENT] %r", msg)
 
 
+@library.register_handler(message.concents.SubtaskResultsSettled)
+def on_subtask_results_settled(msg, **_):
+    """End of UC3. Nothing can be done after this point.
+    I'm either a Provider or Requestor
+    """
+    logger.warning("[CONCENT] End of UC3 by %r", msg)
```
Comment (shadeofblue) [inline golem/network/concent/received_handler.py]: @jiivan I wouldn't use `UC3` in debug messages as I think it's too "niche" -> maybe rather "end of the Force Accept scenario"?

File: golem/network/history.py [hunk 3]
```diff
@@ -324,6 +325,24 @@ def add(msg: message.base.Message,
     )
 
 
+def get(message_class_name: str, task_id: str, subtask_id: str)\
```
Comment (shadeofblue) [inline golem/network/history.py:328]: :+1:

File: golem/network/concent/received_handler.py [hunk 4]
```diff
@@ -202,6 +210,33 @@ def on_force_report_computed_task(self, msg, **_):
...
         )
 
+    @handler_for(message.concents.ForceSubtaskResults)
+    def on_force_subtask_results(self, msg, **_):
+        """I'm a Requestor
+
+        Concent sends his own ForceSubtaskResults with AckReportComputedTask
+        provided by a provider.
+        """
+        sra = history.get('SubtaskResultsAccepted', msg.task_id, msg.subtask_id)
```
Comment (shadeofblue) [inline golem/network/concent/received_handler.py:220]: I really think we should be looking up those messages also using the node id ...

**Review by shadeofblue — APPROVED** [review]
generally, okay - added comments

File: golem/network/concent/received_handler.py (hunk 4 above)
Comment (jiivan) [reply golem/network/concent/received_handler.py:220]: It's doable but I think we should also update all calls to `golem.task.tasksession.get_task_message()` too.  Created an issue #2670

**Pull request merged**